### ISSUE TITLE
ManagementClient.QueueExistsAsync() is working correctly

### ISSUE THREAD

**Issue by SeanFeldman** [Azure/azure-service-bus-dotnet#519]
Give the following code:

```c#
await client.QueueExistsAsync("error");
```
**Actual**
when a connection string w/o Manage rights is used, the client will throw `UnothorizedException` in case `error` queue exists. If the queue doesn't exist, it will return `false`.

**Expected**
Throw whenever connection string has no Manage rights.

This is a potential security issue as information about entities could be revealed.

Note: have not tested with other entities. Could be the same issue.

**Comment by SeanFeldman** [comment]
The old client is throwing `UnauthorizedAccessException`. For consistency, it would be good to throw the same exception as well.

**Comment by nemakam** [comment]
Old client throws `System.UnauthorizedAccessException` . New client throws `Microsoft.Azure.ServiceBus.UnauthorizedException`. 
Wanted all known exceptions to extend from `ServiceBusException`. Hence the change

**Comment by SeanFeldman** [comment]
@nemakam name wise, np. But the issue goes beyond that. Could you please comment on that? Thanks.

**Comment by nemakam** [comment]
Yes. Its a very valid issue. That definitely needs to be fixed soon..
